microsoft/testsuites/vm_extensions/common.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Any
+
+from lisa import (
+    Node,
+)
+from lisa.environment import Environment
+from lisa.operating_system import CpuArchitecture
+from lisa.sut_orchestrator import AZURE
+from lisa.sut_orchestrator.azure.common import (
+    AZURE_SHARED_RG_NAME,
+    AzureNodeSchema,
+    generate_blob_sas_token,
+    get_or_create_storage_container,
+    get_storage_account_name,
+    get_storage_credential,
+)
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.sut_orchestrator.azure.tools import Waagent
+from lisa.util import SkippedException
+from semver import VersionInfo
+
+
+class CommandInfo(object):
+    def __init__(
+        self,
+        file_name: str,
+        expected_exit_code: int,
+    ) -> None:
+        self.command = f"ls '{file_name}'"
+        self.expected_exit_code = expected_exit_code
+        if expected_exit_code == 0:
+            self.failure_message = (
+                f"File {file_name} was not created on the test machine"
+            )
+        else:
+            self.failure_message = (
+                f"File {file_name} downloaded on test machine though it should not have"
+            )
+
+
+def verify_waagent_version_supported(node: Node, environment: Environment) -> None:
+    platform = environment.platform
+    assert isinstance(platform, AzurePlatform)
+
+    env_information = platform.get_environment_information(environment)
```

Comment:
I think you can just use WaAgent tool to get the version. It doesn't need to get it from environment information.